tendrl/commons/flows/create_cluster/utils.py

```diff
@@ -185,65 +185,42 @@ def gluster_create_ssh_setup_jobs(parameters, skip_current_node=False):
 def acquire_node_lock(parameters, flow_name):
     # check job is parent or child
     job = Job(job_id=parameters['job_id']).load()
+    p_job_id = None
     if "parent" in job.payload:
-        # child job not required node lock
-        return
+        p_job_id = job.payload['parent']
 
-    Event(
-        Message(
-            job_id=parameters['job_id'],
-            flow_id=parameters['flow_id'],
-            priority="info",
-            publisher=NS.publisher_id,
-            payload={
-                "message": "Check nodes not locked by other Job"
-            }
-        )
-    )
     fail = False
-    locked_nodes = []
     for node in parameters['Node[]']:
         key = "/nodes/%s/locked_by" % node
         try:
-            job_info = NS._int.client.read(key).value
-            fail = True
-            locked_nodes.append(node)
-            job_info = job_info.encode('utf8').replace("'", "\"")
-            job_info = json.loads(job_info)
-            Event(
-                Message(
-                    job_id=parameters['job_id'],
-                    flow_id=parameters['flow_id'],
-                    priority="error",
-                    publisher=NS.publisher_id,
-                    payload={
-                        "message": "Node %s is already locked by a job %s" % \
-                            (
-                                node,
-                                job_info['job_id']
-                            )
-                    }
-                )
-            )
+            lock_owner_job = NS._int.client.read(key).value            
+            # Do not fail since the parent job has locked the node
+            if p_job_id == job_info:
+                fail = False
+            else:
+                raise FlowExecutionFailedError("Node %s is already locked by a job %s" % (node, lock_owner_job)
```

Comment:
here problem if user chooses nodes A,B ,C which is already locked by some job, this is occur only for node A in UI. But B and C also locked by some job. next time used assumed ok only A is already locked so if he use B,C,D then message in UI will come like B is already locked only